src/pconn.cc

```diff
@@ -227,6 +227,11 @@ IdleConnList::pop()
         if (fd_table[theList_[i]->fd].timeoutHandler == nullptr)
             continue;
 
+        // the cache_peer has been removed from the configuration
+        // TODO: remove all such connections at once during reconfiguration
+        if (theList_[i]->toGoneCachePeer())
```

Comment:
This currently should not happen because the only caller is Icap::ServiceRep::getIdleConnection() - there should be no peers in that context.

Reply:
I agree that we should filter in both methods, even though one of the methods does not currently see connections with CachePeers.